### PR TITLE
Make addBlankLineBefore/After idempotent

### PR DESCRIPTION
**Pull request by gnodet**
## Summary

- `addBlankLineBefore()` and `addBlankLineAfter()` now check whether a blank line already exists before adding one, preventing double blank lines
- Adds `hasBlankLine()` helper that checks for consecutive `lineEnding` sequences in whitespace

## Context

When `PomEditor.insertElementAtPosition()` inserts an element with blank line markers in the ordering config (e.g., `<pluginManagement>` into `<build>`), it calls `addBlankLineAfter()`. If the next sibling (`<plugins>`) already has its normal `\n`-prefixed whitespace, the unconditional `lineEnding` prepend creates a double blank line. Tools like Spotless then flag this as a formatting violation.

This was discovered during Maven 4 compatibility testing: `mvnup apply` adds `<pluginManagement>` entries to POM files, and Sling projects using `spotless-maven-plugin` would fail because the modified POM had double blank lines between `</pluginManagement>` and `<plugins>`.

## Test plan

- [x] New `testAddBlankLineBeforeIsIdempotent` — verifies calling `addBlankLineBefore` on an element that already has a blank line before it is a no-op
- [x] New `testAddBlankLineAfterIsIdempotent` — verifies calling `addBlankLineAfter` on an element that already has a blank line after it is a no-op
- [x] All existing blank line tests still pass (adding blank lines where none exist)
- [x] Full test suite: 316 tests pass, 0 failures

_Claude Code on behalf of Guillaume Nodet_

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced blank-line insertion logic to be idempotent and prevent duplicate blank lines. The system now intelligently inspects existing whitespace patterns before adding new lines, ensuring predictable formatting behavior when operations are repeated on the same content.

* **Tests**
  * Added test cases to verify idempotent behavior of blank-line insertion helpers.

[Review Change Stack](https://app.coderabbit.ai/change-stack/maveniverse/domtrip/pull/238?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: core/src/main/java/eu/maveniverse/domtrip/Editor.java [hunk 1]
```diff
@@ -2091,23 +2091,35 @@ private boolean transferWhitespaceAndRemove(
...
+                if (!hasBlankLine(ws)) {
+                    nextSibling.precedingWhitespace(lineEnding + ws);
+                }
             }
         }
     }
 
+    private boolean hasBlankLine(String whitespace) {
+        return whitespace != null && whitespace.contains(lineEnding + lineEnding);
+    }
```
[the remark this answers is not in the document]
Comment (gnodet) [reply core/src/main/java/eu/maveniverse/domtrip/Editor.java:2130]: Fixed in 8ec3977. Updated `hasBlankLine` to split on `lineEnding` and check interior segments for whitespace-only content, so lines like `"\n    \n"` are correctly recognized as existing blank lines. Also added two idempotency tests for whitespace-only blank lines.

_Claude Code on behalf of Guillaume Nodet_
